engine/combat/utility/sos_appraisal.py

```diff
@@ -303,6 +307,16 @@ def execute_action_complete(self: Self) -> None:
         self.complete = True
         logger.debug("Action Complete")
 
+    def _fallback_for_missing_action_complete(self: Self) -> None:
```

Comment:
This is a bit of a complicated one (especially the conditional), so a docstring would be appreciated.